src/AlternativesPerformance.cpp

```diff
@@ -0,0 +1,142 @@
+      if (this->findAlt(element.first) == false) {
+        throw std::invalid_argument(
+            "The alternatives in the map should be present "
+            "in the performance table.");
+      }
+    }
+    alt_assignment_ = alt_assignment;
+  }
+}
+
+AlternativesPerformance::AlternativesPerformance(
+    int nb_of_perfs, Criteria &crits, std::string prefix,
+    std::map<std::string, std::string> alt_assignment)
+    : PerformanceTable(nb_of_perfs, crits, prefix) {
+  if (mode_ != "alt") {
+    throw std::domain_error(
+        "Performance table mode should be alt to assign default categories.");
+  }
+  if (alt_assignment == std::map<std::string, std::string>{{"", ""}}) {
+    for (std::vector<Perf> pv : pt_) {
+      std::string altName = pv[0].getName();
+      alt_assignment_[altName] = "";
+    }
+  } else {
+    // Check if the alternatives are in the performance table
+    for (std::pair<std::string, std::string> element : alt_assignment) {
+      if (this->findAlt(element.first) == false) {
+        throw std::invalid_argument(
+            "The alternatives in the map should be present "
+            "in the performance table.");
+      }
+    }
+    alt_assignment_ = alt_assignment;
+  }
+}
+
+AlternativesPerformance::AlternativesPerformance(
+    const PerformanceTable &perf_table,
+    std::map<std::string, std::string> alt_assignment)
+    : PerformanceTable(perf_table) {
+  if (alt_assignment == std::map<std::string, std::string>{{"", ""}}) {
+    if (mode_ != "alt") {
+      throw std::domain_error("Performance table mode should be alt to "
+                              "assign default categories.");
+    }
+    for (std::vector<Perf> pv : pt_) {
+      std::string altName = pv[0].getName();
+      alt_assignment_[altName] = "";
+    }
+  } else {
+    // Check if the alternatives are in the performance table
+    for (std::pair<std::string, std::string> element : alt_assignment) {
+      if (this->findAlt(element.first) == false) {
+        throw std::invalid_argument(
+            "The alternatives in the map should be present "
+            "in the performance table.");
+      }
+    }
+    alt_assignment_ = alt_assignment;
+  }
+}
+
+AlternativesPerformance::AlternativesPerformance(
+    const AlternativesPerformance &alt)
+    : PerformanceTable(alt) {
+  alt_assignment_ = alt.alt_assignment_;
+}
+
+AlternativesPerformance::~AlternativesPerformance() {}
+
+std::ostream &operator<<(std::ostream &out,
+                         const AlternativesPerformance &alt) {
+  out << "AlternativesPerformance( PerformanceTable[ ";
+  for (std::vector<Perf> p : alt.pt_) {
+    out << "Performance: ";
+    for (Perf perf : p) {
+      out << perf << " ";
+    }
+    out << "| ";
+  }
+  out << "], AlternativesAssignment{ ";
+
+  auto it = alt.alt_assignment_.begin();
+  while (it != alt.alt_assignment_.end()) {
+    out << it->first << "->" << it->second << " ";
+    it++;
+  }
+  out << "}";
+  return out;
+}
```

Comment:
Do we need to view the performance table when we are dealing with the AA type ?

Reply:
Might be useful to check the results?
But we could do another function like "printAlternativesAssignments" to show only the AA